File: solutions/csharp/hamming/1/Hamming.cs
public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        int hammingDistance = 0;
        char [] first = firstStrand.ToCharArray();
        char [] second = secondStrand.ToCharArray();

        if(firstStrand.Length != secondStrand.Length){
            throw new ArgumentException("strands must be of equal length");
        }
        for(int i = 0; i < firstStrand.Length; i++){
            if(first[i] != second[i]){
                hammingDistance++;
            }
        }
        return hammingDistance;
    }
}

File: solutions/go/hamming/1/hamming.go
package hamming
import "errors"

func Distance(a, b string) (int, error) {
	hammingDistance := 0

    if len(a) != len(b){
        return 0, errors.New("the length of dna strands are not equal")
    }
    for indx, strand := range a {
        if strand != rune(b[indx]) {
            hammingDistance ++
        }
    }

    return hammingDistance, nil
}
